apm-agent-plugins/apm-struts-plugin/src/main/java/co/elastic/apm/agent/struts/ActionProxyAdvice.java

```diff
@@ -19,23 +19,56 @@
 package co.elastic.apm.agent.struts;
 
 import co.elastic.apm.agent.impl.GlobalTracer;
+import co.elastic.apm.agent.impl.transaction.Outcome;
+import co.elastic.apm.agent.impl.transaction.Span;
 import co.elastic.apm.agent.impl.transaction.Transaction;
 import co.elastic.apm.agent.util.TransactionNameUtils;
+import com.opensymphony.xwork2.ActionContext;
 import com.opensymphony.xwork2.ActionProxy;
 import net.bytebuddy.asm.Advice;
 
+import javax.annotation.Nullable;
+
 import static co.elastic.apm.agent.impl.transaction.AbstractSpan.PRIO_HIGH_LEVEL_FRAMEWORK;
 
 public class ActionProxyAdvice {
 
     @Advice.OnMethodEnter(suppress = Throwable.class, inline = false)
-    public static void setTransactionName(@Advice.This ActionProxy actionProxy) {
+    public static Object onEnterExecute(@Advice.This ActionProxy actionProxy) {
         Transaction transaction = GlobalTracer.get().currentTransaction();
         if (transaction == null) {
+            return null;
+        }
+
+        String className = actionProxy.getAction().getClass().getSimpleName();
+        String methodName = actionProxy.getMethod();
+        if (ActionContext.getContext().get("CHAIN_HISTORY") != null) {
+            Span span = transaction.createSpan().withType("struts").withSubtype("action-chain");
```

Comment:
We are currently trying to align span type and sub-types with the spec defined in `span_types.json`, thus I think it would be better to use `span.type` = `app` and `span.subtype` = `internal` when task execution is delegated.

Reply:
Done!